test/tests/functional/pbs_preemption.py

```diff
@@ -251,3 +255,80 @@ def test_preempt_retry(self):
         self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
         self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+
+    def test_vnode_resource_contention(self):
+        """
+        Test to make sure that preemption happens when the resource in
+        contention is vnode.
+        """
+
+        a = {'resources_available.ncpus': 2}
+        self.server.create_vnodes(name='vnode', attrib=a, num=11,
+                                  mom=self.mom, usenatvnode=False)
+
+        a = {'Resource_List.select': '1:ncpus=2+1:ncpus=2'}
+        for each in range(5):
+            j = Job(TEST_USER, attrs=a)
+            jid = self.server.submit(j)
+            self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        # Make sure vnode[4] has jobs running on it
```

Comment:
is it guaranteed that vnode[4] will be job-busy? you created 11 vnodes with 2 ncpus each, so 1 vnode should be free after 5 jobs requesting 4 ncpus, couldn't that be vnode[4] ?

Reply:
Well, there is no node sort key, so unless we change the default configs I think vnode[4] should be used.

Reply:
This kind of makes it seem like the test is testing some functionality by checking that vnode[4] is job-busy. How about just finding the first node that is job-busy and using that?

Reply:
How about I write a comment that randomly select one of the busy nodes and request that vnode in the high priority job?